apps/api/src/server/repositories/document.repository.js

```diff
@@ -22,7 +22,29 @@ export const upsert = (document) => {
 export const getById = (documentGuid) => {
 	return databaseConnector.document.findUnique({
 		where: {
-			guid: documentGuid
+			guid_isDeleted: {
+				guid: documentGuid,
+				isDeleted: false
+			}
+		}
+	});
+};
+
+/**
+ * Get a document by documentGuid and caseId
+ *
+ * @param {string} documentGuid
+ * @param {number} caseId
+ * @returns {import('@prisma/client').PrismaPromise<import('@pins/api').Schema.Document |null>}
+ */
+export const getByIdRelatedToCaseId = (documentGuid, caseId) => {
+	return databaseConnector.document.findFirst({
+		where: {
+			guid: documentGuid,
+			isDeleted: false,
```

Comment:
Great to see you've remembered this! Maybe overkill but I'm wondering if something like this would be a more predictable approach? https://www.prisma.io/docs/concepts/components/prisma-client/middleware/soft-delete-middleware#step-3-optionally-prevent-readupdate-of-soft-deleted-records

Essentially, when we access 'document' collection, we're only allowed to access documents which haven't been deleted. What are your thoughts?

Reply:
this is perfect thanks!

Reply:
I will make those changes today